src/tests/unit/jobs-dao.test.js

```diff
@@ -80,4 +81,137 @@ describe('Test jobs-dao', () => {
       return result.should.eventually.be.rejectedWith(expected);
     });
   });
+
+  const handleJobTestCases = [
+    {
+      message: 'handleJob should call graduateJob without studentEmployeeInd',
+      dbReturn: { rows: [{ count: 1 }], outBinds: { result: null } },
+      expected: null,
+      postBody: { changeReason: { code: 'dummy' } },
+      expectedCalledType: jobStubType.graduateJob,
+    },
+    {
+      message: 'handleJob should call graduateJob',
+      dbReturn: { rows: [{ count: 1 }], outBinds: { result: null } },
+      expected: null,
+      postBody: { studentEmployeeInd: false, changeReason: { code: 'dummy' } },
+      expectedCalledType: jobStubType.graduateJob,
+    },
+    {
+      message: 'handleJob should call studentJob',
+      dbReturn: { rows: [{ count: 1 }], outBinds: { result: null } },
+      expected: null,
+      postBody: { studentEmployeeInd: true, positionNumber: 'C50236', changeReason: { code: 'dummy' } },
+      expectedCalledType: jobStubType.studentJob,
+    },
+    {
+      message: 'handleJob should call terminateJob for student',
+      dbReturn: { rows: [{ count: 1 }], outBinds: { result: null } },
+      expected: null,
+      postBody: { studentEmployeeInd: true, positionNumber: 'C50236', changeReason: { code: 'TERMJ' } },
+      expectedCalledType: jobStubType.terminateJob,
+    },
+    {
+      message: 'handleJob should call terminateJob for graduate',
+      dbReturn: { rows: [{ count: 1 }], outBinds: { result: null } },
+      expected: null,
+      postBody: { studentEmployeeInd: false, positionNumber: 'C50236', changeReason: { code: 'TERMJ' } },
+      expectedCalledType: jobStubType.terminateJob,
+    },
+    {
+      message: 'handleJob should call updateLaborChangeJob for graduate',
+      dbReturn: { rows: [{ count: 1 }], outBinds: { result: null } },
+      expected: null,
+      postBody: { studentEmployeeInd: false, positionNumber: 'C50236', changeReason: { code: 'NONE' } },
+      expectedCalledType: jobStubType.updateLaborChangeJob,
+    },
+    {
+      message: 'handleJob should call updateLaborChangeJob for student',
+      dbReturn: { rows: [{ count: 1 }], outBinds: { result: null } },
+      expected: null,
+      postBody: { studentEmployeeInd: true, positionNumber: 'C50236', changeReason: { code: 'NONE' } },
+      expectedCalledType: jobStubType.updateLaborChangeJob,
+    },
+  ];
+  _.forEach(handleJobTestCases, ({
+    message,
+    dbReturn,
+    expected,
+    postBody,
+    expectedCalledType,
+  }) => {
+    it(message, async () => {
+      const stubGraduateJob = sinon.stub(contrib, 'graduateJob');
+      const stubStudentJob = sinon.stub(contrib, 'studentJob');
+      const stubTerminateJob = sinon.stub(contrib, 'terminateJob');
+      const stubUpdateLaborChangeJob = sinon.stub(contrib, 'updateLaborChangeJob');
+      const daoProxy = createDaoProxy(dbReturn, dbReturn);
```

Comment:
Why do we want to pass `dbReturn` twice to `createDaoProxy`?

Reply:
Because handleJobs makes multiple calls to connection.execute and I need to specify the required data to continue the execution of the code. The two get jobs tests use dbReturn for the first call and  "{ rows: [{}] }" for second and beyond calls to connection.execute. The handleJobs tests needs to use dbReturn for all calls.

Reply:
If you want to redesign it to be a function with consecutive returns, I think the better way is to make the input become an array and put all your returns in it in order. That way, the function would have more flexibility (could have more than 2 returns, if needed) and is also easier to understand. For this change, I would also add a document string to make it clear.

Reply:
The refactoring for change was minimal.

Reply:
What do you mean? I think code quality and readability are very important and NOT minimal.

Reply:
I meant the refactoring of tests to adapt to the change of allowing a variable length of DB results.